giddy/directional.py

```diff
@@ -324,25 +324,73 @@ def _calc(self, Y, w, k):
         results['dy'] = dy
         return results
 
-    def plot(self, attribute=None):
+    def plot(self, attribute=None, ax=None, **kwargs):
         """Plot the rose diagram.
 
         Parameters
         ----------
         attribute : (n,) ndarray, optional
             Variable to specify colors of the colorbars.
-
+        ax : Matplotlib Axes instance, optional
+            If given, the figure will be created inside this axis.
+            Default =None. Note, this axis should have a polar projection.
+        **kwargs : keyword arguments, optional
+            Keywords used for creating and designing the plot.
+            Note: 'c' and 'color' cannot be passed when attribute is not None
+
+        Returns
+        -------
+        fig : Matplotlib Figure instance
+            Moran scatterplot figure
+        ax : matplotlib Axes instance
+            Axes in which the figure is plotted
+        
+        Examples
+        --------
+        Imports
+        >>> import geopandas as gpd
+        >>> import pandas as pd
+        >>> import libpysal.api as lp
+        >>> from libpysal import examples
+        >>> import numpy as np
+        >>> import matplotlib.pyplot as plt
+        >>> from giddy.directional import Rose
+        get csv and shp files
+        >>> shp_link = examples.get_path('us48.shp')
+        >>> df = gpd.read_file(shp_link)
+        >>> income_table = pd.read_csv(examples.get_path("usjoin.csv"))
+        calculate relative values
+        >>> for year in range(1969, 2010):
+        ...     income_table[str(year) + '_rel'] = (
+        ...         income_table[str(year)] / income_table[str(year)].mean())  
+        merge to one gdf
+        >>> gdf = df.merge(income_table,left_on='STATE_NAME',right_on='Name')
+        retrieve spatial weights and data for two points in time
+        >>> w = lp.Queen.from_dataframe(gdf)
+        >>> w.transform = 'r'
+        >>> y1 = gdf['1969_rel'].values
+        >>> y2 = gdf['2000_rel'].values
+        calculate rose Object
+        >>> Y = np.array([y1, y2]).T
+        >>> rose = Rose(Y, w, k=5)
+        plot
+        >>> fig1, _ = rose.plot()
+        >>> plt.show()
+        customize plot
+        >>> fig, _ = rose.plot()(attribute=y1)
+        >>> plt.show()
         """
         use_splot = False
         try:
             import splot.giddy
             use_splot = True
         except ImportError:
-            warnings.warn('To get better results, please install the `splot` package',
-                          UserWarning)
+            warnings.warn('This method relies on importing `splot` in future',
+                          DeprecationWarning)
 
         if use_splot:
```

Comment:
if the import fails, then you'll see the warning in line 388, and then this will fail on a name error.

This happens because `use_splot` isn't defined by the time you hit this if.

I think you can solve this if you set `use_splot=False` after line 389

Reply:
I am not sure about the current model of `splot` in terms of supporting visualization for other pysal packages such as `giddy` and `esda`. My thought was that `giddy` would only focus on analytics while leaving visualization to `splot`, meaning that `giddy` would not have a plot function but users can easily import `splot` when visualization is needed. We can also demonstrate the visualization functionalities provided by `splot` in jupyter notebook gallery. (`splot` and `geopandas` will be included in the `requirement_dev.txt` to support sphinx building) In this way, `giddy` will be a lightweight package.

Reply:
@weikang9009 the idea was that the plotting functionality remains in `splot` and can be called as a `.plot()` method inside other sub packages. This makes plotting specific objects more user friendly. I will change the import/ install requirement of `splot` to make it a soft-dependency so `giddy` remains a lightweight package with a plotting option if needed.

Reply:
I apologize to @weikang9009 as I didn't convey the splot based discussions that led to the this pr and its potential implications for giddy. 

Not sure this creates any weight for giddy as splot is treated as a soft dependency here. 

I am +1 on adding the plot method as I think it makes sense for the designers of giddy to suggest the default analytical plots. Again, it is based on a soft dependency so we don't increase the install requirements, but give the interested user warnings if they try the plot method without splot installed.

Reply:
Thanks for the clarification.